### PR TITLE
fix: [PL-29982]: Throw exception in case secret is not found

### PR DESCRIPTION
**Pull request by nishantsaini20**
## Describe your changes
With the availability of force delete feature there can be cases where a secret has been forcefully deleted even when it being referred by existing connector. In such a case null value was returned which results in NullPointerException or incorrect credentials error, without giving any useful error message for the case.
Whenever encryption details are being fetched and any secret is not found during the flow will now result in secret not found exception.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40659)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java [hunk 1]
```diff
@@ -767,6 +768,9 @@ && isSecretIdentifierAPathReference(secretIdentifier)) {
                                            .fieldName(field.getName())
                                            .build());
             }
+          } else {
+            throw new SecretManagementException(
+                SECRET_NOT_FOUND, String.format("Secret [%s] not found or has been deleted.", secretIdentifier), USER);
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGEncryptedDataServiceImpl.java:773]: should we add accountId, orgId etc. to logs?

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40659)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by piyush-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40659)* status: :shipit: complete! ✓

**Comment by nishantsaini20** [comment]
trigger codebasehash

**Pull request merged**